packages/@aws-cdk/aws-ec2/lib/machine-image.ts

```diff
@@ -150,6 +175,77 @@ export class GenericSSMParameterImage implements IMachineImage {
   }
 }
 
+/**
+ * Properties for GenericSsmParameterImage
+ */
+export interface SsmParameterImageOptions {
+  /**
+   * Operating system
+   *
+   * @default OperatingSystemType.LINUX
+   */
+  readonly os?: OperatingSystemType;
+
+  /**
+   * Custom UserData
+   *
+   * @default - UserData appropriate for the OS
+   */
+  readonly userData?: UserData;
+
+  /**
+   * Whether the AMI ID is cached to be stable between deployments
+   *
+   * By default, the newest image is used on each deployment. This will cause
+   * instances to be replaced whenever a new version is released, and may cause
+   * downtime if there aren't enough running instances in the AutoScalingGroup
+   * to reschedule the tasks on.
+   *
+   * If set to true, the AMI ID will be cached in `cdk.context.json` and the
+   * same value will be used on future runs. Your instances will not be replaced
+   * but your AMI version will grow old over time. To refresh the AMI lookup,
+   * you will have to evict the value from the cache using the `cdk context`
+   * command. See https://docs.aws.amazon.com/cdk/latest/guide/context.html for
+   * more information.
+   *
+   * Can not be set to `true` in environment-agnostic stacks.
+   *
+   * @default false
+   */
+  readonly cachedInContext?: boolean;
```

Comment:
I think just "cache" would be fine

Reply:
I want people thinking about "context" though...

Reply:
Personally, this feels like an implementation detail that doesn't need to be surfaced in the API name. All the customer cares is that the AMI does not change automatically without cache eviction

Reply:
I agree with you in general. What concerns me is that users might not know where to look for said cache eviction--which is why I want that word in there.